### ISSUE TITLE
This directory is not available, please check the log files or contact the administrator.

### ISSUE THREAD

**Issue by DasCanard** [nextcloud/deck#3292]
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments. 

**Describe the bug**
After Updating Deck to version 1.5.1 i will get this error when i am sharing files
"This directory is not available, please check the log files or contact the administrator."

**To Reproduce**
1. Share a folder
2. Open the URL and see the "This directory is not available, please check the log files or contact the administrator." error

**Screenshots**
![image](https://user-images.githubusercontent.com/17070204/132094254-03afd9ac-2787-4fa5-9ca8-fbdc6c5728f8.png)

**Client details:**
 - OS: Windows 10
 - Browser: Edge
 - Version: 22.1.1
 - Device: Desktop

## Server configuration detail

**Operating system:** Linux 4.15.0-154-generic #161-Ubuntu SMP Fri Jul 30 13:04:17 UTC 2021 x86_64

**Webserver:** nginx/1.21.1 (fpm-fcgi)

**Database:** mysql 10.5.12

**PHP version:** 8.0.10

Modules loaded: Core, date, libxml, openssl, pcre, zlib, filter, hash, json, Reflection, SPL, session, standard, sodium, cgi-fcgi, mysqlnd, PDO, xml, apcu, bcmath, bz2, calendar, ctype, curl, dom, mbstring, FFI, fileinfo, ftp, gd, gettext, gmp, iconv, igbinary, imagick, intl, ldap, exif, mysqli, pdo_mysql, Phar, posix, readline, redis, shmop, SimpleXML, smbclient, sockets, sysvmsg, sysvsem, sysvshm, tokenizer, xmlreader, xmlwriter, xsl, zip, libsmbclient, Zend OPcache

**Nextcloud version:** 22.1.1 - 22.1.1.2

**Updated from an older Nextcloud/ownCloud or fresh install:** 

**Where did you install Nextcloud from:** unknown

<details><summary>Signing status</summary>

Array
(
)

</details>

<details><summary>List of activated apps</summary>

```
Enabled:
 - accessibility: 1.7.0
 - activity: 2.15.0
 - admin_audit: 1.11.0
 - calendar: 2.3.3
 - checksum: 1.1.2
 - circles: 22.1.1
 - cloud_federation_api: 1.4.0
 - comments: 1.11.0
 - contacts: 4.0.2
 - contactsinteraction: 1.2.0
 - dashboard: 7.1.0
 - dav: 1.18.0
 - deck: 1.5.1
 - documentserver_community: 0.1.11
 - extract: 1.3.2
 - federatedfilesharing: 1.11.0
 - federation: 1.11.0
 - files: 1.16.0
 - files_pdfviewer: 2.3.0
 - files_rightclick: 1.1.0
 - files_sharing: 1.13.2
 - files_trashbin: 1.11.0
 - files_versions: 1.14.0
 - files_videoplayer: 1.11.0
 - forms: 2.3.0
 - logreader: 2.7.0
 - lookup_server_connector: 1.9.0
 - mail: 1.10.3
 - metadata: 0.14.0
 - nextcloud_announcements: 1.11.0
 - notes: 4.1.1
 - notifications: 2.10.1
 - oauth2: 1.9.0
 - password_policy: 1.12.0
 - photos: 1.4.0
 - privacy: 1.6.0
 - provisioning_api: 1.11.0
 - qownnotesapi: 21.7.0
 - recommendations: 1.1.0
 - richdocuments: 4.2.2
 - richdocumentscode: 6.4.1004
 - serverinfo: 1.12.0
 - settings: 1.3.0
 - sharebymail: 1.11.0
 - spreed: 12.1.1
 - support: 1.5.0
 - systemtags: 1.11.0
 - text: 3.3.0
 - theming: 1.12.0
 - twofactor_backupcodes: 1.10.1
 - twofactor_totp: 6.1.0
 - updatenotification: 1.11.0
 - user_status: 1.1.1
 - viewer: 1.6.0
 - weather_status: 1.1.0
 - workflow_pdf_converter: 1.7.0
 - workflowengine: 2.3.1
Disabled:
 - encryption
 - files_external
 - firstrunwizard
 - maps
 - officeonline
 - polls
 - survey_client
 - user_ldap

```
</details>

<details><summary>Configuration (config/config.php)</summary>

```
{
    "passwordsalt": "***REMOVED SENSITIVE VALUE***",
    "secret": "***REMOVED SENSITIVE VALUE***",
    "trusted_domains": [
        "cloud",
        "dascanard.cloud",
        "upload.dascanard.cloud"
    ],
    "datadirectory": "***REMOVED SENSITIVE VALUE***",
    "dbtype": "mysql",
    "version": "22.1.1.2",
    "overwrite.cli.url": "https:\/\/dascanard.cloud",
    "dbname": "***REMOVED SENSITIVE VALUE***",
    "dbhost": "***REMOVED SENSITIVE VALUE***",
    "dbport": "",
    "dbtableprefix": "oc_",
    "mysql.utf8mb4": true,
    "dbuser": "***REMOVED SENSITIVE VALUE***",
    "dbpassword": "***REMOVED SENSITIVE VALUE***",
    "installed": true,
    "instanceid": "***REMOVED SENSITIVE VALUE***",
    "activity_expire_days": 14,
    "auth.bruteforce.protection.enabled": true,
    "blacklisted_files": [
        ".htaccess",
        "Thumbs.db",
        "thumbs.db"
    ],
    "cron_log": true,
    "default_phone_region": "DE",
    "enable_previews": true,
    "enabledPreviewProviders": [
        "OC\\Preview\\PNG",
        "OC\\Preview\\JPEG",
        "OC\\Preview\\GIF",
        "OC\\Preview\\BMP",
        "OC\\Preview\\XBitmap",
        "OC\\Preview\\Movie",
        "OC\\Preview\\PDF",
        "OC\\Preview\\MP3",
        "OC\\Preview\\TXT",
        "OC\\Preview\\MarkDown"
    ],
    "filesystem_check_changes": 0,
    "filelocking.enabled": "true",
    "simpleSignUpLink.shown": false,
    "htaccess.RewriteBase": "\/",
    "integrity.check.disabled": false,
    "knowledgebaseenabled": false,
    "log_rotate_size": 104857600,
    "logfile": "\/daten\/nextcloud.log",
    "loglevel": 0,
    "logtimezone": "Europe\/Berlin",
    "memcache.local": "\\OC\\Memcache\\APCu",
    "memcache.locking": "\\OC\\Memcache\\Redis",
    "preview_max_x": 1920,
    "preview_max_y": 1080,
    "preview_max_scale_factor": 1,
    "redis": {
        "host": "***REMOVED SENSITIVE VALUE***",
        "port": 0,
        "timeout": 0
    },
    "quota_include_external_storage": false,
    "share_folder": "\/Freigaben",
    "skeletondirectory": "",
    "trashbin_retention_obligation": "auto, 90",
    "maintenance": false,
    "updater.release.channel": "stable",
    "theme": "",
    "mail_smtpmode": "smtp",
    "mail_smtpsecure": "tls",
    "mail_sendmailmode": "smtp",
    "mail_from_address": "***REMOVED SENSITIVE VALUE***",
    "mail_domain": "***REMOVED SENSITIVE VALUE***",
    "mail_smtpauthtype": "LOGIN",
    "mail_smtpauth": 1,
    "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
    "mail_smtpport": "587",
    "mail_smtpname": "***REMOVED SENSITIVE VALUE***",
    "mail_smtppassword": "***REMOVED SENSITIVE VALUE***"
}
```
</details>

**External storages:** files_external is disabled

**Encryption:** no

**User-backends:** 
 * OC\User\Database

**Talk configuration:** 

STUN servers
 * no custom server configured

TURN servers
 * no custom server configured

Signaling servers (mode: internal):
 * no custom server configured

**Browser:** Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/92.0.4515.159 Safari/537.36 Edg/92.0.902.84

**Comment by apg1980** [comment]
for me as well

**Comment by stefan-niedermann** [comment]
@apg1980 Please use only the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue if you don't have any further information. This clutters issues and makes them harder to read, understand and solve.

**Comment by piratenpanda** [comment]
Same here, issue with my config reported here https://help.nextcloud.com/t/nextcloud-shares-not-working-this-directory-is-unavailable/123100

**Comment by yostral** [comment]
It appears also with the folders shared with another NC instance.

An user from instance A shares a folder with a user on instance B, the user on instance B can't access the shared folder anymore.

**Comment by stefanroesler** [comment]
We have the same bug.

**Comment by apg1980** [comment]
I have just created an public share, imediately the error appears.

**Comment by robertskiba** [comment]
Same issue here. Only workaround was to disable the deck app!

**Comment by mist** [comment]
I manually applied the changes from #3299 and it fixes the problem.

**Comment by mejo-** [comment]
I also experienced this issue and can confirm it's fixed by applying the patch from #3299. Probably it should be backported to `stable22`, no?

**Comment by RJ-Make** [comment]
Same issue here.

**Comment by labcerouno** [comment]
Same here. Disabling Deck made the trick.

**Comment by juliushaertl** [comment]
Fixes have been released as part of the latest release 1.5.2 / 1.4.4
